backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/Migrations.java

```diff
@@ -0,0 +1,38 @@
+package org.molgenis.emx2.sql;
+
+import java.io.IOException;
+import org.molgenis.emx2.MolgenisException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Migrations {
+  private static Logger logger = LoggerFactory.getLogger(Migrations.class);
+
+  public static void initOrMigrate(SqlDatabase db) {
+    int version = MetadataUtils.getVersion(db.getJooq());
+
+    // migration steps
+    if (version < 0) MetadataUtils.init(db.getJooq());
+    if (version < 1)
+      executeMigrationFile(db, "migration1.sql", "upgraded MOLGENIS.version_metadata");
```

Comment:
say for example de DB version is at version 3 and the CODE is at version 7 , this if , if , if, if setup does not scale nice . I would suggest a small function the determines the migrations needed ( if ) any , logs the result ( this is important info for maintainers, and then start to run the migration, logging what migration is run and the status ( success or fail ). DB version should probably be set between each migration ( say you go form 3 to 7 and migration step 6 fails , i would expect the db version the set to 5 ).

Reply:
The idea is that you need all migrations, always. So empty database runs all migration. That ensures migrations work exactly same as the 'create from empty'.

If it fails, I believe migration should roll back completely so you can go back to previous software version that worked for you instead of being left hanging halfway. Importantly, this tx was missing so added.